### PR TITLE
JSUI-3282 Added specific documentOpen analytics for SmartSnippet and SmartSnippetSuggestions

### PR DESCRIPTION
**Pull request by btaillon**
https://coveord.atlassian.net/browse/JSUI-3282

Demo: https://ypc9n.csb.app/

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: src/ui/ResultLink/ResultLink.ts [hunk 1]
```diff
@@ -224,7 +224,9 @@ export class ResultLink extends Component {
      */
     onClick: ComponentOptions.buildCustomOption<(e: Event, result: IQueryResult) => any>(() => {
       return null;
-    })
+    }),
+
+    logAnalytics: ComponentOptions.buildCustomOption<(href: string) => void>(() => null)
```
Comment (olamothe) [inline src/ui/ResultLink/ResultLink.ts:229]: This new option should probably be documented, as well as unit tested.

File: src/ui/SmartSnippet/SmartSnippetCollapsibleSuggestion.ts [hunk 2]
```diff
@@ -186,4 +196,18 @@ export class SmartSnippetCollapsibleSuggestion {
       this.checkbox.el
     );
   }
+
+  private sendOpenSourceAnalytics(element: HTMLElement, href: string) {
+    return this.bindings.usageAnalytics.logCustomEvent<IAnalyticsSmartSnippetSuggestionOpenSourceMeta>(
```
Comment (olamothe) [inline src/ui/SmartSnippet/SmartSnippetCollapsibleSuggestion.ts:201]: You should be able to do just: 
```suggestion
    return this.usageAnalytics.logCustomEvent<IAnalyticsSmartSnippetSuggestionOpenSourceMeta>(
```

**Review by olamothe — APPROVED** [review]
I would arrange for a "demo" build (uploaded to a hosted search page in dev, maybe) and to verify with Oscar if everything he needs is there.

File: src/ui/SmartSnippet/SmartSnippetCollapsibleSuggestion.ts (hunk 2 above)
Comment (btaillon) [reply src/ui/SmartSnippet/SmartSnippetCollapsibleSuggestion.ts:201]: CollapsibleSuggestion doesn't extend `Component`.

**Review by ThibodeauJF — APPROVED** [review]

**Review by samisayegh — APPROVED** [review]

File: src/ui/Analytics/AnalyticsActionListMeta.ts [hunk 3]
```diff
@@ -1460,6 +1490,20 @@ export var analyticsActionCauseList = {
...
+   *
+   * ```javascript
+   * {
+   *  actionCause: "openSmartSnippetSource",
+   *  actionType: "smartSnippet"
+   * }
+   * ```
+   */
+  openSmartSnippetSource: <IAnalyticsActionCause>{
+    name: 'openSmartSnippetSource',
```
[the remark this answers is not in the document]
Comment (btaillon) [reply src/ui/Analytics/AnalyticsActionListMeta.ts:1504]: I chose it

**Comment by btaillon** [comment]
@operecoveo Here's a demo of those new analytics: https://ypc9n.csb.app/

**Review by dlutzCoveo — APPROVED** [review]

**Pull request merged**